core/src/test/java/de/evoila/cf/broker/controller/utils/DashboardUtilsTest.java

```diff
@@ -0,0 +1,767 @@
+package de.evoila.cf.broker.controller.utils;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Arrays;
+
+import de.evoila.cf.broker.model.DashboardClient;
+import de.evoila.cf.broker.model.catalog.Dashboard;
+import de.evoila.cf.broker.model.catalog.ServiceDefinition;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@SuppressWarnings("InnerClassMayBeStatic")
+@ExtendWith(MockitoExtension.class)
+class DashboardUtilsTest {
+
+    private static final String     HAPPY_DASHBOARD_URL = "https://www.test.com";
+    private static final String     HAPPY_INSTANCE_ID   = "975834d6-bb9f-4825-98f2-ddaad251a6d3";
+    private static final String     HAPPY_REDIRECT_URI  = "https://www.test2.com";
+    private static final String[]   HAPPY_APPENDIXES    = {"first",
+                                                           "second",
+                                                           "third"};
+
+    @Mock
+    private ServiceDefinition serviceDefinition;
+    @Mock
+    private Dashboard dashboard;
+
+    @Nested
+    class hashDashboard {
+
+        @Test
+        void isTrue() {
+            DashboardClient dashboardClient = mock(DashboardClient.class);
+            when(serviceDefinition.getDashboard())
+                    .thenReturn(dashboard);
+            when(dashboard.getUrl())
+                    .thenReturn(HAPPY_DASHBOARD_URL);
+            when(serviceDefinition.getDashboardClient())
+                    .thenReturn(dashboardClient);
+            boolean result = DashboardUtils.hasDashboard(serviceDefinition);
+            assertTrue(result);
+        }
+
+        @Nested
+        class isFalse {
+
+            private boolean result;
+
+            @AfterEach
+            void tearDown() {
+                assertFalse(result);
```

Comment:
I repeatedly saw assertions in @AfterEach methods. Is this common, because I would not expect an assertion here.
Also it is dangerous to put it here since tracing down the actual failed methods is not ready doable. The following example shows me forcing a test to fail:

"org.opentest4j.AssertionFailedError: 
Expected :false
Actual   :true
<Click to see difference>
	at de.evoila.cf.broker.controller.utils.DashboardUtilsTest$hashDashboard$isFalse.tearDown(DashboardUtilsTest.java:59)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)"

It is not obvious which method failed its assertion. So please either add messages to the assertions or call the assertion out of the test methods.